apps/backend/src/modules/users/user.service.ts

```diff
@@ -45,6 +47,19 @@ class UserService implements Service {
 		};
 	}
 
+	public async findByEmail(email: string): Promise<UserDto> {
+		const user = await this.userRepository.findByEmail(email);
+
+		if (!user) {
+			throw new AuthError({
+				message: ErrorMessage.INCORRECT_CREDENTIALS,
+				status: HTTPCode.UNAUTHORIZED,
+			});
+		}
```

Comment:
findByEmail should not throw this error, it should either return the user or null